src/components/Editor/Preview/ReactPopup.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { Component } from "react";
+
+import "./ReactPopup.css";
+
+class ReactPopup extends Component {
+  props: {
+    roots: Array<Object>
+  };
+
+  renderComponentAttributes(root: Object) {
+    if (!("preview" in root.contents.value)) {
+      return <span className="attrKey">none</span>;
+    }
+    const attributes = root.contents.value.preview.ownProperties;
+
+    return Object.keys(attributes).map(key => {
+      const val = attributes[key].value;
+      let content = "";
+
+      switch (typeof val) {
```

Comment:
this might be moved into its own function, and instead have this:

```javascript
function getContent(val) {
  switch (typeof val) {
  case "object":
    if (val.class === "Function") {
      return`${val.displayName}()`;
      // .. and so on
}

const content = getContent(val);
```